### ISSUE TITLE
Rule S1854 should not report false positives with lambdas and other scope changing constructs

### ISSUE THREAD

**Issue by tvsonar** [SonarSource/sonarlint-vs#319]
```
var fieldsAssigned = ImmutableHashSet<IFieldSymbol>.Empty;

analysisContext.RegisterSyntaxNodeAction(c =>
{    
    var fieldSymbol = ...
    fieldsAssigned = fieldsAssigned.Add(fieldSymbol);

}, SyntaxKind.SimpleAssignmentExpression);
```

PR #11
